model.py

```diff
@@ -9,43 +9,98 @@
 import utils
 
 
+class GaussianAttention(nn.Module):
+    def __init__(self, query_size=5):
+        super(GaussianAttention, self).__init__()
+        self.linear = nn.Linear(in_features=query_size, out_features=2)
+
+    def forward(self, query, embedded_text, text_mask, position):
+        """query should be of shape [seq_len, batch_size, query_size]]
+        textEmbedding should be of shape [text_len, batch_size]
+        text_mask should be of shape [text_len, batch_size]
+        position should be of shape [seq_len, batch_size] (or None)
+        Returns tuple of:
+          * a tensor of shape [seq_len, batch_size, embedding_dim]
+          * a tensor of shape [seq_len, batch_size] containing positions."""
+        # TODO(neil): rewrite so that we don't need seq_len leading dimension
+        masked_embedding = embedded_text * text_mask.unsqueeze(-1)
+        seq_len = query.shape[0]
+        batch_size = query.shape[1]
+        if position is None:
+            position = torch.zeros((seq_len, batch_size), device=query.device)
+        query_result = torch.exp(self.linear(query))
+        assert query_result.shape == (seq_len, batch_size, 2)
+        assert embedded_text.shape[1] == batch_size
+        mean, std = query_result.unbind(dim=-1)
+
+        # We divide by 20 because avg number of strokes/character is 20.
+        mean = position + mean.cumsum(dim=0)/20
+        # TODO(neil): Why these extra dimensions with value?
+        indices = torch.arange(0, embedded_text.shape[0], device=query.device).view(1, embedded_text.shape[0], 1, 1)
+        weights = torch.exp(-std.view(seq_len, 1, batch_size, 1) * (mean.view(seq_len, 1, batch_size, 1) - indices) ** 2)
+        weightedSum = masked_embedding * weights
+        new_context = weightedSum.sum(dim=1)
+
+        return new_context, mean
```

Comment:
Think about what do you want to return? I think that `mean` is not the right return here.